### ISSUE TITLE
Error "task.apply(...).then is not a function" on node 9.7.1

### ISSUE THREAD

**Issue by alangibson** [SGrondin/bottleneck#55]
When I run this on node 9.7.1

```
const Bottleneck = require("bottleneck");

const limiter = new Bottleneck({
  maxConcurrent: 1,
  minTime: 1000
});

limiter.schedule(() => 'im the call')
  .then((result) => { console.log('handle result', result) });

```

I get

```
/[...]/node_modules/bottleneck/lib/Bottleneck.js:376
          return task.apply({}, args).then(function (...args) {
                                      ^

TypeError: task.apply(...).then is not a function
    at Object.wrapped ([...]/node_modules/bottleneck/lib/Bottleneck.js:376:39)
    at Timeout._executing.(anonymous function).timeout.setTimeout [as _onTimeout] ([..]/node_modules/bottleneck/lib/Bottleneck.js:233:32)
    at ontimeout (timers.js:466:11)
    at tryOnTimeout (timers.js:304:5)
    at Timer.listOnTimeout (timers.js:267:5)
```

Is there a known issue with node 9.7.1?

**Comment by SGrondin** [comment]
It's not an issue with Node 9.7.1, your code will fail on any version of Node.

From the README:
>Simply put, schedule takes a function Fn and a list of arguments. **Fn must return a promise.** schedule returns a promise that will be executed according to the rate limits.

The function you are giving to `.schedule()` does not return a Promise, it returns a String.

Try this instead:
```js
limiter.schedule(() => Promise.resolve('im the call'))
  .then((result) => { console.log('handle result', result) });
```
Future versions of Bottleneck might wrap your non-Promise values into a Promise, but that could lead to surprising behavior in some cases.

**Comment by SGrondin** [comment]
Please reopen if the error persists 😃

**Comment by alangibson** [comment]
The dangers of late night programming :man_facepalming: I think just including that use of Promise in one of the examples in the README would probably be enough for people like me that tend to read the code and skim the prose.

**Comment by SGrondin** [comment]
I'll see if I can improve the Readme!

**Comment by SGrondin** [comment]
In the newly released `2.2.1` you'd get the following error message:

>The function given to `schedule()` did not return a Promise. You may need to return `Promise.resolve(data)`. You returned: This is a string (string)'

I've also added this to the Readme:

-----------------

If your function does not return a promise, it needs to use `Promise.resolve` like so:

```js
// GOOD
limiter.schedule(() => Promise.resolve("This is a string"))
.then(data => console.log(data));

// INCORRECT!
limiter.schedule(() => "This is a string")
.then(data => console.log(data));
```
